lib/insights/api/common/filter.rb

```diff
@@ -111,55 +111,55 @@ def integer(k, val)
         def comparator_contains(key, value)
           return value.each { |v| comparator_contains(key, v) } if value.kind_of?(Array)
 
-          self.query = query.where(arel_table[key].matches("%#{query.sanitize_sql_like(value)}%", nil, true))
+          self.query = query.where("#{key} LIKE ?", "%#{ActiveRecord::Base.sanitize_sql_like(value)}%")
         end
 
         def comparator_contains_i(key, value)
           return value.each { |v| comparator_contains_i(key, v) } if value.kind_of?(Array)
 
-          self.query = query.where(arel_table[key].lower.matches("%#{query.sanitize_sql_like(value.downcase)}%", nil, true))
+          self.query = query.where("#{key} ILIKE ?", "%#{ActiveRecord::Base.sanitize_sql_like(value)}%")
         end
 
         def comparator_starts_with(key, value)
-          self.query = query.where(arel_table[key].matches("#{query.sanitize_sql_like(value)}%", nil, true))
+          self.query = query.where("#{key} LIKE ?", "#{ActiveRecord::Base.sanitize_sql_like(value)}%")
         end
 
         def comparator_starts_with_i(key, value)
-          self.query = query.where(arel_table[key].lower.matches("#{query.sanitize_sql_like(value.downcase)}%", nil, true))
+          self.query = query.where("#{key} ILIKE ?", "#{ActiveRecord::Base.sanitize_sql_like(value)}%")
         end
 
         def comparator_ends_with(key, value)
-          self.query = query.where(arel_table[key].matches("%#{query.sanitize_sql_like(value)}", nil, true))
+          self.query = query.where("#{key} LIKE ?", "%#{ActiveRecord::Base.sanitize_sql_like(value)}")
         end
 
         def comparator_ends_with_i(key, value)
-          self.query = query.where(arel_table[key].lower.matches("%#{query.sanitize_sql_like(value.downcase)}", nil, true))
+          self.query = query.where("#{key} ILIKE ?", "%#{ActiveRecord::Base.sanitize_sql_like(value)}")
         end
 
         def comparator_eq(key, value)
-          self.query = query.where(key => value)
+          self.query = query.where(key => Array(value).collect { |i| ActiveRecord::Base.sanitize_sql(i) })
         end
 
         def comparator_eq_i(key, value)
-          values = Array(value).map { |v| query.sanitize_sql_like(v.downcase) }
+          values = Array(value).collect { |i| ActiveRecord::Base.sanitize_sql(i.downcase) }
 
-          self.query = query.where(arel_table[key].lower.matches_any(values))
+          self.query = query.where("lower(#{key}) IN (?)", values)
```

Comment:
If we're interpolating the key here, should we use something like this to sanitize it? https://www.rubydoc.info/gems/pg/PG/Connection#quote_ident-class_method

(this applies generally to this PR)

Reply:
The only things that ever get this far are documented attributes.  Do you think we still need to quote them?